StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift

```diff
@@ -522,12 +522,30 @@ extension STPAPIClient: FinancialConnectionsAPIClient {
             shouldPollAccounts,
             !linkedAccountIds.isEmpty
         {
-            return pollAccountNumbersForSelectedAccounts(
+            let promise = Promise<(
+                manifest: FinancialConnectionsSessionManifest,
+                customSuccessPaneMessage: String?
+            )>()
+            pollAccountNumbersForSelectedAccounts(
                 linkedAccountIds: linkedAccountIds
             )
-            .chained { _ in
-                return saveAccountsToLinkHandler()
+            .observe { result in
+                switch result {
+                case .success:
+                    saveAccountsToLinkHandler()
+                        .observe { result in
+                            promise.fullfill(with: result)
+                        }
+                case .failure(let error):
+                    self.disableNetworking(
+                        disabledReason: "account_numbers_not_available",
+                        clientSecret: clientSecret
+                    ).observe { _ in } // ignoring return is intentional
+
+                    promise.reject(with: error)
+                }
             }
+            return promise
```

Comment:
the purpose of this modification is to add an API call to `disableNetworking` IF `pollAccountNumbersForSelectedAccounts` fails

...everything else should remain the same (if this is implemented correctly)

[web code](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/fc4b14ad/src/linkedAccounts/inner/hooks/v3/useSaveToNetworkAndLink.ts#L117-L119)